ledger-tool/src/main.rs

```diff
@@ -2244,6 +2254,15 @@ fn main() {
                     },
                 );
 
+                let snapshot_archive_format = arg_matches
+                    .value_of("snapshot_archive_format")
+                    .map_or(ArchiveFormat::TarZstd, |s| {
+                        s.parse::<ArchiveFormat>().unwrap_or_else(|e| {
+                            eprintln!("Error: {}", e);
+                            exit(1)
+                        })
+                    });
```

Comment:
I think this `parse()` will always fail here without something equivalent to what's in `validator/src/main.rs`:

https://github.com/solana-labs/solana/blob/68800982eac507314f51189a30d39f35eca403ba/validator/src/main.rs#L2654-L2663

So minimally the validator/main code needs to be updated, and something equivalent needs to be implemented here. Alternatively (and probably better) is to move this impl into archive_format.rs. The `ArchiveFormat::try_from()` can be updated to support both name and extension strings.

i.e.

```rust
    fn try_from(extension: &str) -> Result<Self, Self::Error> {
        match extension {
            TAR_BZIP2_EXTENSION | "bz2" => Ok(ArchiveFormat::TarBzip2),
            TAR_GZIP_EXTENSION | "gzip" => Ok(ArchiveFormat::TarGzip),
            TAR_ZSTD_EXTENSION | "zstd" => Ok(ArchiveFormat::TarZstd),
            TAR_EXTENSION | "tar" | "none" => Ok(ArchiveFormat::Tar),
            _ => Err(ParseError::InvalidExtension),
        }
    }
```

or a new fn, like `fn from_cli_arg(...)`, or something similar should be fine too. This may be possible/simplified from the strum crate as well (but I haven't checked).

Reply:
yes. fixed and updated in the recent commit.